src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -5288,6 +5288,32 @@ protected sealed override ISymbol GetEnclosingSymbolCore(int position, Cancellat
             return this.GetEnclosingSymbol(position, cancellationToken);
         }
 
+        private protected sealed override ImmutableArray<IImportScope> GetImportScopesCore(int position, CancellationToken cancellationToken)
+        {
+            position = CheckAndAdjustPosition(position);
+            var binder = GetEnclosingBinder(position);
+            var builder = ArrayBuilder<IImportScope>.GetInstance();
+
+            for (var chain = binder?.ImportChain; chain != null; chain = chain.ParentOpt)
+            {
+                var imports = chain.Imports;
+                if (imports.IsEmpty)
+                    continue;
+
+                Debug.Assert(imports.Usings.All(static u => u.UsingDirectiveReference != null));
+
+                // Try to create a node corresponding to the imports of the next higher binder scope. Then create the
+                // node corresponding to this set of imports and chain it to that.
+                builder.Add(new SimpleImportScope(
+                    imports.UsingAliases.SelectAsArray(static kvp => kvp.Value.Alias.GetPublicSymbol()),
```

Comment:
>UsingAliases [](http://example.com/codeflow?start=28&length=12)

It doesn't look like the order is going to be deterministic.

Reply:
Resolved. API review is that these will not be deterministic.  API is doc'ed as such, and code is implemented to try to enforce no common order so as to break any product code that assumes that.